### PR TITLE
added authentication 

### PR DESCRIPTION
**Pull request by vidhi-thakur**
- created a login page
- redirecting to login page if the user access the private private page
- after login, redirecting user back to original page user tried to access
- fixed side bar scroll issue

Live URL: https://temp-notes.netlify.app/

### CODE REVIEW

File: src/context/authContext.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useState } from "react";
+
+const authContext = createContext();
+
+export const useAuth = () => useContext(authContext)
+
+export const AuthProvider = ({ children }) => {
+    const [isUserLoggenIn, setIsUserLoggenIn] = useState(false)
```
Comment (LordUttam) [inline src/context/authContext.js]: spelling mistake? 'd' in logged

**Review by LordUttam — APPROVED** [review]
There are some spelling errors like loggenIn and sidear. Other than that everything looks good to me.

File: src/context/authContext.js (hunk 1 above)
Comment (vidhi-thakur) [reply src/context/authContext.js]: thanks for noticing, updated the change

File: src/pages/login/Login.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const updatePassword = event => {
+        setPassword(event.target.value)
+    }
+    const loginUser = (e) => {
+        e.preventDefault()
+        if(userName !== '' && password !== '') {
+            updateLoginState()
+            navigate(location?.state?.from?.pathname)
+        }
+    }
```
Comment (tushartiwari7) [inline src/pages/login/Login.js:26]: Liked that you have taken care of empty values.
To add on you can also show alerts or error toasts for a better UX.

**Review by tushartiwari7 — APPROVED** [review]
The Code seems very modular and easy to read.
Taken care of very small implications as well, liked that.
Wasn't able to review UI as link isn't there.
But liked the Code, Great work @vidhi-thakur

File: src/pages/login/Login.js (hunk 2 above)
Comment (vidhi-thakur) [reply src/pages/login/Login.js:26]: Thanks for your review. Have added the deployment URL in the description, if you'd like to check. 

Would definitely add in next itreation
>To add on you can also show alerts or error toasts for a better UX.

**Pull request merged**